datafusion/physical-plan/src/lib.rs

```diff
@@ -117,14 +117,27 @@ pub mod udaf {
 /// [`required_input_ordering`]: ExecutionPlan::required_input_ordering
 pub trait ExecutionPlan: Debug + DisplayAs + Send + Sync {
     /// Short name for the ExecutionPlan, such as 'ParquetExec'.
-    fn name(&self) -> &'static str {
+    fn name(&self) -> &'static str
+    where
+        Self: Sized,
+    {
+        Self::static_name()
+    }
```

Comment:
Now we can't call `plan.name()` inside `ExecutionPlanVisitor`:
```rust
    fn pre_visit(&mut self, plan: &dyn ExecutionPlan) -> Result<bool, Self::Error> {
        let plan_name = plan.name().to_owned();
        ...
    }
```

```
     |
1552 |         let plan_name = plan.name().to_owned();
     |                              ^^^^
     |
     |
121  |         Self: Sized,
     |               ----- this has a `Sized` requirement
```

Reply:
Yes, this method cannot be called from a trait object.

I'm considering making a breaking API change to deprecate the default implementation and provide the static name on each impl. This can avoid the issue we're facing now and also keeps other functionalities works like before.

Reply:
I think it's nice to be able to call `name()` on the dyn object.
As long as we can keep doing that I would be happy 😄

Reply:
Great, I'll submit a patch later this day (or tomorrow).

Reply:
FYI https://github.com/apache/datafusion/pull/11047